packages/make-styles/src/runtime/reduceToClassNameForSlots.ts

```diff
@@ -0,0 +1,55 @@
+import { DEFINITION_LOOKUP_TABLE } from '../constants';
+import { hashSequence } from './utils/hashSequence';
+import { CSSClassesMapBySlot, CSSClassesMap, CSSClasses } from '../types';
+
+/**
+ * Reduces a classname map for slot to a classname string. Uses classnames according to text directions.
+ *
+ * @private
+ */
+export function reduceToClassName(classMap: CSSClassesMap, dir: 'ltr' | 'rtl'): string {
+  let className = '';
+
+  // eslint-disable-next-line guard-for-in
+  for (const propertyHash in classMap) {
+    const classNameMapping: CSSClasses = classMap[propertyHash];
+
+    if (classNameMapping) {
+      const hasRTLClassName = Array.isArray(classNameMapping);
```

Comment:
just curious - what is the motivation for not using the array always? wouldn't it make the LTR scenario slightly faster?

Reply:
> just curious - what is the motivation for not using the array always? wouldn't it make the LTR scenario slightly faster?

If we will use arrays always we will get additional characters always:

```
rules: [
  // 1️⃣ with array always
  ['.foo{color:red}'], // two more chars on each rule
  // vs.
  // 2️⃣ with strings for LTR only rules
  '.foo{color:red}'
]
```

Even according to our fixtures there is a small amount of rules that flip.

---

> wouldn't it make the LTR scenario slightly faster?

Honestly I would rather try to confirm it with benchmarks 🙄 From my POV less objects/array is better (https://www.youtube.com/watch?v=ff4fgQxPaO0).

Reply:
#weneedbenchmarks